gcloud/datastore/connection.py

```diff
@@ -415,3 +417,20 @@ def _add_keys_to_request(request_field_pb, key_pbs):
     for key_pb in key_pbs:
         key_pb = _prepare_key_for_request(key_pb)
         request_field_pb.add().CopyFrom(key_pb)
+
+
+def _parse_commit_response(commit_response_pb):
+    """Extract response data from a commit response.
+
+    :type commit_response_pb: :class:`._datastore_pb2.CommitResponse`
+    :param commit_response_pb: The protobuf response from a commit request.
+
+    :rtype: tuple
+    :returns': The pair of the number of index updates and a list of
+               :class:`._entity_pb2.Key` for each incomplete key that was
+               completed in the commit.
+    """
+    mut_result = commit_response_pb.mutation_result
+    index_updates = mut_result.index_updates
+    completed_keys = list(mut_result.insert_auto_id_key)
+    return index_updates, completed_keys
```

Comment:
1. `Connection` has always been a low-level API so there is no reason to drop the data returned
1. The index updates give an idea of how much quota an app is using (e.g. how much money it is eating) and there are certainly some callers that may use it. (The only high-level caller that use `Connection.commit` is `Batch`, which discards the `index_updates`.)